.github/workflows/nightly.yml

```diff
@@ -106,6 +60,18 @@ jobs:
           cd ./packages/react-native/sdks/hermes || exit 1
           . ./utils/build-apple-framework.sh
           build_host_hermesc_if_needed
+      - name: Upload HermesC Artifact
+        uses: actions/upload-artifact@v4.3.1
+        with:
+          name: hermesc-apple
+          path: ./packages/react-native/sdks/hermes/build_host_hermesc
+      - name: Cache hermesc apple
+        uses: actions/cache/save@v4.0.0
+        if: ${{ github.ref == 'refs/heads/main' }} # To avoid that the cache explode.
+        with:
+          path: ./packages/react-native/sdks/hermes/build_host_hermesc
+          key: v2-hermesc-apple-${{ needs.prepare_hermes_workspace.outputs.hermes-version }}-${{ needs.prepare_hermes_workspace.outputs.react-native-version }}
+          enableCrossOsArchive: true
```

Comment:
apply the same cache-and-artifacts optimization we have in hermew-workspace to stabilize HermesC